datafusion/core/tests/dataframe.rs

```diff
@@ -124,6 +125,39 @@ async fn sort_on_unprojected_columns() -> Result<()> {
     Ok(())
 }
 
+#[tokio::test]
```

Comment:
I suggest to add those sql in comment into `sqllogicaltest`.
In this test, we can `data-derive-test`

Reply:
I also suggest add test in `datafusion/sql/tests/integration_test.rs`.
In this test, we can see the `Plantree` of plan to check those sql

Reply:
same comments with @jackwener . It is better to add e2e test for this case in the `distinct` test or the `order by` test.